src/test/java/com/epam/ta/page/google/GoogleComputingEnginePage.java

```diff
@@ -104,8 +108,18 @@ public GoogleComputingEnginePage fillForm() {
         waitGoogleComputingEngineElement(select2x375);
         select2x375.click();
         selectCenterLocation.click();
-        waitGoogleComputingEngineElement(selectFrankfurt);
-        selectFrankfurt.click();
+
+        switch (TestDataReader.getTestData("place")) {
+            case "frankfurt":
+                waitGoogleComputingEngineElement(selectFrankfurt);
+                selectFrankfurt.click();
+                break;
+            case "osaka":
+                waitGoogleComputingEngineElement(selectOsaka);
+                selectOsaka.click();
+                break;
+        }
```

Comment:
Pages should contain locators and methods which perform actions with locators. Move logic to test or service layer